rovercode/drivers/grove_motors.py

```diff
@@ -1,43 +1,66 @@
-except ImportError:
-    LOGGER.warning("Grove I2C motor driver lib unavailable. Using dummy.")
-    from drivers.dummy_grovepi_interface import motor_driver
+if os.getenv('DEVELOPMENT', 'false').lower() == 'true':
+    LOGGER.warning("In DEVELOPMENT mode. Using dummy.")
+    from drivers.dummy_grovepi_interface import MiniMotorDriver
+else:
+    import sys
+    sys.path.append('../GrovePi/Software/Python')
+    from grove_mini_motor_driver.grove_mini_motor_driver import MiniMotorDriver  # noqa
 
 
 class GroveMotors:
     """Helper functions for setting the GrovePi motor controllers."""
 
-    PORT_A = 'a'
-    PORT_B = 'b'
-    DIRECTION_FORWARD = 0b10
-    DIRECTION_BACKWARD = 0b01
+    DIRECTION_FORWARD = 'FORWARD'
+    DIRECTION_BACKWARD = 'REVERSE'
+    LEFT_DRIVER_ADDRESS = 0x60
+    RIGHT_DRIVER_ADDRESS = 0x65
+    DIRECTIONS = (DIRECTION_FORWARD, DIRECTION_BACKWARD)
 
     def __init__(self):
         """Create an instance of motor interface."""
-        self.interface = motor_driver()
-        self.speeds = {self.PORT_A: 0, self.PORT_B: 0}
-        self.directions = {self.PORT_A: 0, self.PORT_B: 0}
-
-    def set_speed(self, port, speed, direction):
-        """Set the speed of the motors."""
-        port, direction = port.lower(), direction.lower()
-        if self.speeds[port] == speed and self.directions[port] == direction:
-            return
-        self.speeds[port], self.directions[port] = speed, direction
+        try:
+            self.interface = MiniMotorDriver(self.LEFT_DRIVER_ADDRESS,
+                                             self.RIGHT_DRIVER_ADDRESS)
+        except RuntimeError:
+            # MiniMotorDriver can fail based on hardware platform on init
+            LOGGER.warning("Grove I2C mini motor driver lib unavailable."
+                           "Using dummy.")
+            from drivers.dummy_grovepi_interface import MiniMotorDriver \
+                as DummyMiniMotorDriver
+            self.interface = DummyMiniMotorDriver(
+                self.LEFT_DRIVER_ADDRESS, self.RIGHT_DRIVER_ADDRESS)
+        self.interface.stopMotors()
+
+    def set_left_speed(self, speed, direction):
+        """Set the speed of the left motors."""
+        self._set_speed('left', speed, direction)
 
-        self.interface.MotorSpeedSetAB(self.speeds[self.PORT_A],
-                                       self.speeds[self.PORT_B])
+    def set_right_speed(self, speed, direction):
+        """Set the speed of the right motor."""
+        self._set_speed('right', speed, direction)
 
-        direction_bits = \
-            (self.directions[self.PORT_A] << 2) + \
-            (self.directions[self.PORT_B])
-        self.interface.DirectionSet(direction_bits)
+    def _set_speed(self, motor, speed, direction):
+        if direction not in self.DIRECTIONS:
+            LOGGER.error("Invalid direction %s", direction)
+            return
+        if speed > 100:
+            LOGGER.warning("Capping speed %s to 100", speed)
```

Comment:
You need to actually change `speed` here to cap it

Reply:
Done!